### PR TITLE
refactor: added custom exceptions when required parameters are missing

### PR DESCRIPTION
**Pull request by manchuck**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: packages/auth/lib/auth.ts [hunk 1]
```diff
@@ -28,44 +34,86 @@ export class Auth implements AuthInterface {
...
+      this.privateKey,
+      this.jwtOptions,
+    )}`;
   };
 
   createSignatureHash = async <T>(
     params: AuthSignedParams & T,
   ): Promise<AuthSignedParams & T> => {
     log('Creating signature hash');
+    if (!this.apiKey) {
```
Comment (dragonmantank) [inline packages/auth/lib/auth.ts:111]: This method should also make sure they supplied a signature hash. If there is no signature string then we can't generate a signature.

**Review by dragonmantank — CHANGES_REQUESTED** [review]
The new changes look OK except I think we should be checking for API key and signature string in the signature method, not just the signature stuff.

This also seems to be built off the Proactive Connect code, can that be stripped out since this PR has nothing to do with PC?

**Comment by manchuck** [comment]
@dragonmantank That was a strange rebase that happened. Either way, some GitFu and PC have been removed

**Review by dragonmantank — APPROVED** [review]
LGTM

**Pull request merged**